languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -34110,6 +34110,125 @@ though, and has since been slightly modified)-->
     </rule>
   </category>
   <category id="CAT_TYPOGRAPHIE" name="Typographie" type="typographical">
+      <rulegroup id="VIRGULE_DEBUT_DE_PHRASE" name="virgule debut de phrase" default="temp_off">
+          <rule>
+              <pattern>
+                      <token postag="SENT_START"/>
+                  <marker>
+                      <token regexp="yes">cependant|néanmoins|pourtant|toutefois</token>
+                  </marker>
+                  <token postag="[RAVNJPZCD].*" postag_regexp="yes"/>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\2,</suggestion>
+              <example correction="Cependant,"><marker>Cependant</marker> il ne le remarqua pas.</example>
+              <example correction="Pourtant,"><marker>Pourtant</marker> manger entre les repas est nocif pour la santé.</example>
+              <example correction="Néanmoins,"><marker>Néanmoins</marker> Marie n'en pensait pas moins.</example>
+              <example correction="Toutefois,"><marker>Toutefois</marker> le maire était d'humeur généreuse aujourd'hui.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token>en</token>
+                  <token>conséquence</token>
+                  <token regexp="yes">de|du|des|directe|et</token>
+              </antipattern>
+              <pattern>
+                  <token postag="SENT_START"/>
+                  <token>en</token>
+                  <marker>
+                      <token regexp="yes">effet|conséquence</token>
+                  </marker>
+                  <token postag="[RAVNJPZCD].*" postag_regexp="yes"/>
```

Comment:
Same as above

Reply:
Also, this pattern needs an exception in case the user writes "En effet." or "En effet !"